frontend/src/components/chat/user-journey-dialog.tsx

```diff
@@ -0,0 +1,130 @@
+"use client";
+
+import React, { useState } from "react";
+import { useTheme } from "next-themes";
+import { Plus } from "lucide-react";
+import { getSuggestions } from "@/actions/get-suggestions";
+import {
+  JsonData,
+  JsonEditor,
+  monoDarkTheme,
+  monoLightTheme,
+} from "json-edit-react";
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogFooter,
+} from "@/components/ui/dialog";
+import { Button } from "@/components/ui/button";
+import { SuggestedQuestions } from "@/lib/types";
+
+type UserJourneyDialogProps = {
+  isDialogOpen: boolean;
+  setIsDialogOpen: (open: boolean) => void;
+  setSuggestions: React.Dispatch<
+    React.SetStateAction<SuggestedQuestions | undefined>
+  >;
+};
+
+export function UserJourneyDialog({
+  isDialogOpen,
+  setIsDialogOpen,
+  setSuggestions,
+}: UserJourneyDialogProps) {
+  const defaultUserFlow = [
+    [
+      ["brain_region", "Brain stem"],
+      ["artifact", "Neuron density"],
+    ],
+    [
+      ["brain_region", "Cerebellum"],
+      ["artifact", "Neuron density"],
+    ],
+    [
+      ["brain_region", "Brain stem"],
+      ["artifact", "Electrophysiology"],
+    ],
+  ];
+
+  const [userFlowData, setUserFlowData] = useState(defaultUserFlow);
+  const { theme } = useTheme();
+  const isLightTheme = theme === "light";
+
+  const handleSubmit = async () => {
+    setIsDialogOpen(false);
+    const questions = await getSuggestions(userFlowData);
```

Comment:
So I would still use the `useActionState` since it would give us the `isPending` and you can retrieve the result from `state`.


Regarding the `isPending`, could we have a loader or skeleton or something like that? Not a huge fan of the user clicking a button and then having no visual feedback that it triggered something